### PR TITLE
Update the release page with info from private wiki.

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

I am trying to move more info from our private wiki, into this public webpage.

Changes
=======

Move info about version scheme

Move info about releasing a library and releasing a forked library.

Move info about releasing a product so that the documentation about the release is only in repo + styleguide + salt-master (for the buildbot configuration)

How to try and test the changes
===============================

reviewers: @lesingerouge  @alibotean @brunogola @atykhonov 

check that changes make sense

I was talking during the latest release meeting about documenting our release process... so here I go

page available at http://styleguide.chevah.com/release-process.html

for the .SpecialNNN I was not sure whether to got for [-SpeacialNNN] but I feel that using just the dot as a separator makes things much simpler

we also use .chevah1 for denoting that we done some re-packaging or forking

the actual product  release is not yet fully documented as the documentation is found in

* chevah/server repo for the available paver commands
* chevah/brink repo for buildbot client side tools
* chevah/salt-master for buildmaster configuration
* chevah/styleguide - this info

but at least it is no longer stored in the private wiki :)

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

File: content/release.rst [hunk 1]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
 new release on the website.
 
 
+Release Manager
+===============
+
+During the time of an in-work release a single person is in charge of
+driving the release as is called the release manager.
```
Comment (alibotean) [inline content/release.rst]: as -> and

File: content/release.rst [hunk 2]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
...
+
+Some tasks done by the release manager:
+
+* Define milestone description / due date
+* About 1 week before the release, create a release milestone and move all
+  closed ticket from the 'next-release' milestone to the new release milestone.
```
Comment (alibotean) [inline content/release.rst]: ticket -> tickets

File: content/release.rst [hunk 3]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
...
+* Create a dedicated ticket for the release itself, and associated it
```
Comment (alibotean) [inline content/release.rst]: associated -> associate

File: content/release.rst [hunk 4]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
...
+  to the new release milestone.
+* Make sure the ticket dedicated to the release as an owner and that the
```
Comment (alibotean) [inline content/release.rst]: as -> has

File: content/release.rst [hunk 5]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
...
+  owner will do the required steps.
+* Some tickets from the `next-release` milestone which are not yet closed,
+  but are soon to be closed (like needs_merge) can also be moved to the new
+  release milestone.
+* Check post-commit buildbot results for master (at once per week) to make
+  sure no regression were introduced on the tests executed post-merge.
```
Comment (alibotean) [inline content/release.rst]: regression -> regressions

File: content/release.rst [hunk 6]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
...
+* Create high priority tickets in case the tests are failing on master.
+* Coordinates story tickets for the milestone
```
Comment (alibotean) [inline content/release.rst]: Missing .

File: content/release.rst [hunk 7]
```diff
@@ -50,6 +49,96 @@ The release review request should also include the text used to announce the
...
+
+The Release Branch
+==================
+
+A release branch starts like any other branch by creating a ticket in Trac.
+
+The release branch should be created from master (for latest release) or
+from a tag (for a maintenance release).
+
+To integrate with out automated process the release branch should be named:
```
Comment (alibotean) [inline content/release.rst]: out -> our

File: content/release.rst [hunk 8]
```diff
@@ -231,15 +320,21 @@ soon as possible.
 Each release has a certain overhead, and the overhead should be minimized by
 automating the release process.
 
+PATCH versions are released as soon as a defect is fixed,
+usually one week after it has been initially discovered and reported.
+Security issues have top priority and a fix is released as soon as possible.
+PATHC version don't include any new functionality and changes are focused only
```
Comment (alibotean) [inline content/release.rst]: PATHC -> PATCH

Comment (alibotean) [inline content/release.rst]: version -> version | don't -> doesn't

File: content/release.rst [hunk 9]
```diff
@@ -291,3 +386,92 @@ MAJOR release without any other manual migration process.
...
+* Install the previous major release and use the configuration from the new
+  release to start the product.
+  Make sure that all services are properly configured and no errors are
+  reported.
+
+
+Releasing a forked library
+==========================
+
+Sometimes we might need do small or major changes to an upstream
```
Comment (alibotean) [inline content/release.rst]: to do

File: content/release.rst [hunk 10]
```diff
@@ -291,3 +386,92 @@ MAJOR release without any other manual migration process.
...
+package/library.
+For example changes were were rejected upstream, or not yet released upstream
```
Comment (alibotean) [inline content/release.rst]: ~~were~~

File: content/release.rst [hunk 11]
```diff
@@ -291,3 +386,92 @@ MAJOR release without any other manual migration process.
...
+or just minor re-packaging changes.
+
+The forked versions should be published only on our private PyPI server and
+all versions should use the `.chevahN` suffix.
+
+When forking an upstream project, keep the master/trunk branch as upstream.
+You can created separate branches dedicated to the Chevah project like
```
Comment (alibotean) [inline content/release.rst]: created -> create

**Comment by alibotean** [comment]
changes-approved

File: content/release.rst [hunk 12]
```diff
@@ -291,3 +386,92 @@ MAJOR release without any other manual migration process.
...
+
+A library is a software which provides code shared by multiple products.
+Libraries should always be released using the standard package management
+system.
+
+Releasing a library consists of the following:
+
+* creating a distributable in a format used by the package manager.
+* publishing the distributable to the package manager website.
+  In our case most of the time it will be a Python package pushed to on our
```
Comment (brunogola) [inline content/release.rst]: pushed to **on** our -> pushed to our

File: content/release.rst [hunk 13]
```diff
@@ -291,3 +386,92 @@ MAJOR release without any other manual migration process.
 
 All removal warnings should have a similar format to simplify filtering and
 reporting them.
+
+Here are some steps you can use for testing that the compatibility between
```
Comment (brunogola) [inline content/release.rst]: testing **that** the -> testing the

**Comment by adiroiban** [comment]
@atykhonov @lesingerouge  can you please review this ticket?

Feel free to make your changes and push them.

Thanks!

**Comment by adiroiban** [comment]
thanks for the review...19 days waiting for the review so I am going to merge it as it is.

**Pull request merged**